commands/config.js

```diff
@@ -12,23 +12,106 @@ exports.command = 'config';
-    default: 'table'
-  },
-  out: {
-    description: 'Output to a file'
-  },
-  flatten: {
-    description: 'Flatten the config object into dot notation',
-    type: 'boolean',
-    default: false
-  }
+exports.builder = (yargs) => {
+  const electronPrebuiltVersion = require('electron-prebuilt/package.json').version;
+  const pkg = require('../lib/package');
+  var opts = {
+    verbose: {
+      describe: 'Confused or trying to track down a bug and want lots of debug output?',
+      type: 'boolean',
+      default: false
+    },
+    platform: {
+      describe: 'What platform are we building for?',
+      choices: ['win32', 'linux', 'darwin'],
+      default: process.platform
+    },
+    arch: {
+      describe: 'What platform architecture are we building for?',
+      choices: ['x64', 'x86'],
+      default: process.arch
+    },
+    electron_version: {
+      describe: 'What version of electron are we using?',
+      default: electronPrebuiltVersion
+    },
+    version: {
+      describe: 'What version of the application are we building?',
+      default: process.env.npm_package_version || pkg.version
+    },
+    name: {
+      describe: 'What is the kebab cased name of the application?',
+      default: process.env.npm_package_name || pkg.name
+    },
+    product_name: {
+      describe: 'What is the name of the application we should display to humans?',
+      default: pkg.productName || pkg.name
+    },
+    description: {
+      describe: 'What is the description of the application we should display to humans?',
+      default: process.env.npm_package_description || pkg.description
+    },
+    sign: {
+      describe: 'Should this build be signed?',
+      type: 'boolean',
+      default: true
+    },
+    signtool_params: {
+      describe: 'What extra cli arguments should be passed to signtool.exe?',
+      default: process.env.SIGNTOOL_PARAMS || null
+    },
+    favicon_url: {
+      description: 'A URL to an ICO file to use as the application icon (e.g. Windows: displayed in Control Panel > Programs and Features)',
+      default: _.get(pkg, 'config.hadron.build.win32.favicon_url')
+    },
+    evergreen_revision: {
+      description: 'What revision, aka commit sha1 is evergreen building?',
+      type: 'string',
+      default: process.env.EVERGREEN_REVISION
+    },
+    evergreen_build_variant: {
+      description: 'build_variant on evergreen',
+      type: 'string',
+      default: process.env.EVERGREEN_BUILD_VARIANT
+    },
+    evergreen_branch_name: {
+      description: 'branch_name on evergreen',
+      type: 'string',
+      default: process.env.EVERGREEN_BRANCH_NAME
+    },
+    github_token: {
+      description: 'GitHub API token.',
+      default: process.env.GITHUB_TOKEN
+    },
+    github_owner: {
+      default: pkg.github_owner
+    },
+    github_repo: {
+      default: pkg.github_repo
+    },
+    author: {
+      default: pkg.author || pkg.authors
+    }
+  };
+  _.assign(opts, {
```

Comment:
Should we continue using `_.assign` over `Object.assign`? We tend to get warnings when installing packages, not sure if they are related:

<img width="486" alt="screen shot 2016-12-13 at 10 42 05 pm" src="https://cloud.githubusercontent.com/assets/1217010/21139093/65bb0c08-c185-11e6-9ec9-b4aa2acc7332.png">